plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/AbstractMetastoreTableOperations.java

```diff
@@ -113,7 +117,26 @@ protected final void commitNewTable(TableMetadata metadata)
         Table table = builder.build();
 
         PrincipalPrivileges privileges = owner.map(MetastoreUtil::buildInitialPrivilegeSet).orElse(NO_PRIVILEGES);
-        metastore.createTable(table, privileges);
+        try {
+            metastore.createTable(table, privileges);
+            cleanupOnFailure = false;
+        }
+        catch (SchemaNotFoundException
+               | TableAlreadyExistsException
+               | UnsupportedOperationException e) {
```

Comment:
Same as on `io.trino.plugin.iceberg.catalog.glue.GlueIcebergTableOperations#commitToExistingTable` add a few relevant comments that provide more context about the exception handling logic.

Reply:
commented